File: B_Spreadsheets.py


def i_to_letters(num: int):
    s = ""

    while(num > 0):
        tmp = int((num - 1) % 26)
        if tmp < 0:
            tmp += 26
        s += chr(tmp + ord('A'))
        num //= 26
        
    
    s = s[::-1]
    return s


def letters_to_i(letters: str):
    num, n = 0, len(letters)

    for i in range(n - 1, - 1, -1):
        num += pow(26, n - (i + 1)) * ((ord(letters[i]) - ord('A')) + 1)

    
    return num

tt = int(input())

while tt > 0:
    s = input()
    try:
        if s[0] == 'R' and '1' <= s[1] <= '9':
            i, num = 1, 0
            while s[i] != 'C':
                num = num * 10 + int(s[i])
                i += 1
            
            row = num

            i += 1
            num = 0
            while(i < len(s)):
                num = num * 10 + int(s[i])
                i += 1
            
            col = i_to_letters(num);

            print(col + str(row))


        else:
            i = 0 
            letters = ""
            while(not s[i].isdigit()):
                letters += s[i]
                i += 1

            col = letters_to_i(letters)

            num = 0
            while(i < len(s)):
                num = num * 10 + int(s[i])
                i += 1

            print("R" + str(num) + "C" + str(col))
    except Exception:
        print('Exception: ' + s)

    tt -= 1    

File: B_Spreadsheets.cpp
#include <bits/stdc++.h>
#define int long long
using namespace std;



string i_to_letters(int num) {
    string letters = "";
    int k = 26, l = 1;
    while(k < num) {
        num -= k;
        k *= 26;
        ++l;
    }

    for (int i = 0; i < l; ++i) {
        k /= 26;
        for (int j = 0; j < 26; j++)
        if (num <= k) {
                letters +=  (char ('A' + j));
                break;
        } else num -= k;
	}
    return letters;
}

int letters_to_i(string letters) {
    int num = 0, n = size(letters);

    for(int i = n - 1; i >= 0; --i) {
        num += pow(26, n - (i + 1)) * ((letters[i] - 'A') + 1);
    }

    return num;
}

void solve() {
    string s; cin >> s;

    if(s[0] == 'R' && isdigit(s[1]) && find(s.begin(), s.end(), 'C') != s.end()) {
        int i = 1, num = 0;
        while(s[i] != 'C') {
            num = num * 10 + (s[i] - '0');
            ++i;
        }

        int row = num;

        ++i;
        num = 0;
        while(i < size(s)) {
            num = num * 10 + (s[i] - '0');
            ++i;
        }
        // cout << num << '\n';
        string col = i_to_letters(num);

        cout << col + to_string(row) << '\n';
    } 

    else {
        int i = 0; string letters = "";
        while(!isdigit(s[i])) letters += s[i++];

        int col = letters_to_i(letters);

        int num = 0;
        while(i < size(s)) num = num * 10 + (s[i++] - '0');

        cout << "R" + to_string(num) + "C" + to_string(col) << '\n';
    }

}

signed main() {
    ios_base::sync_with_stdio(0); cin.tie(0); cout.tie(0);


    int tt; cin >> tt;
    while(tt--) {
        solve();
    }



    return 0;
}